src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -452,6 +453,15 @@ public void setInputs( final Map<String, Object> inputs ) {
     if ( inputs.containsKey( DASHBOARD_MODE ) ) {
       dashboardMode = "true".equals( String.valueOf( inputs.get( DASHBOARD_MODE ) ) ); //$NON-NLS-1$
     }
+    if ( inputs.containsKey( QUERY_LIMIT ) ) {
+      try {
+        if ( this.getReport() != null && inputs.get( QUERY_LIMIT ) != null ) {
+          this.getReport().setQueryLimit( Integer.parseInt( (String) inputs.get( QUERY_LIMIT ) ) );
```

Comment:
And here: Check the existing query limit. There may be one on the report already. 

Also: The input here comes from the URL, so it can be any value. You will have to add additional check to ensure that the value you receive from the inputs is within the system- and user-limits set in the configuration. So the value you should set here is the minimum of either of these values (accounting for the meaning of <= 0 as "no limit", of course).